src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2092,6 +2031,168 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 }
 
 
+/*
+ * ProcessWorkerExpressionList is a utility function which processes the
+ * expressions that are intended to be added to the worker target list.
+ *
+ * In summary, the function gets a list of expressions, converts them to target
+ * entries and updates all the necessary fields such that the expression is correctly
+ * added to the worker query's target list.
+ *
+ * Adding an expression to the target list includes generating a target entry,
+ * appending it to the newTargetEntryList and incrementing the projection number.
+ *
+ * If the expression is requested to be added to the group clause, the entry
+ * is appended to groupClauseList and sort group reference is incremented.
+ */
```

Comment:
Could you add descriptions of the other parameters as well? (e.g. `targetProjectionNumber`)

Reply:
How do they look now? I'm planning to add similar comments to #2107 if you're good with them